### ISSUE TITLE
Signal/store wrappers are unable to be properly analyzed.

### ISSUE THREAD

**Issue by tealsnow** [solidjs-community/eslint-plugin-solid#190]
**Describe the bug**
Signal/store wrappers are unable to be properly analyzed.

**To Reproduce**
Using a wrapper around a signal or store results in the lint warning `For proper analysis, array destructuring should be used to capture the first result of this function call.`. Such as [@solid-primitives/storage](https://primitives.solidjs.community/package/storage)'s `makePerisisted`:

```tsx
const SomeComponent = () => {
  const [state, setState] = makePersisted(createSignal(false));
  // ...
}
```

**Expected behavior**
The linter should understand the either these custom wrappers are destructured intelligently via the destructuring into `[name, setName]` via the wrapper. Or allow a configuration options to remove ambiguity for what does result in a store/signal like object.

**Environment:**

- OS: Linux
- Node version: `v20.19.4`
- `eslint-plugin-solid` version: `0.14.5`
- `eslint` version: `9.35.0`

**Additional context**
Related issue here: https://github.com/solidjs-community/eslint-plugin-solid/issues/84 with another comment explaining the same issue: https://github.com/solidjs-community/eslint-plugin-solid/issues/84#issuecomment-2958180149

I think the best solution here is to allow the user to specify what counts as a signal/store wrapper, as I foresee that attempting to intelligently work out what is or isn't to be a deep rabbit hole without any perfect solution. 

- [x] I would be willing to contribute a PR to fix this issue
I'm perfectly willing to draft up a PR to fix this, just LMK with a comment, and I'll look at it when I have time.

**Comment by ryansolid** [comment]
Confirmed on current main: wrapper patterns like `makePersisted(createSignal(false))` still report `shouldDestructure` against the inner call. The rule should look through call wrappers when the result is itself destructured. Related: #193. On the backlog.

**Comment by ryansolid** [comment]
Fixed in 0.18.0. The rule saw the inner `createSignal(false)` call, noticed *its own* result wasn't destructured (it's an argument to `makePersisted`), and warned — even though the wrapper's result is destructured right there. Primitive calls passed through wrapper functions are now analyzed as if assigned to the outermost wrapper's captured result, which both kills the false positive and keeps real reactivity analysis working through wrappers like `makePersisted` (an untracked setup-level read of the wrapped signal still warns, as it should). Uncaptured wrapper results stay silent.
